### PR TITLE
Report built-in modiles in kmod.available, fixes #43945

### LINKED ISSUE #43945: kmod.present doesn't work with compiled-in modules
### Description of Issue/Question

If module is compiled in the kernel, `kmod.present` fails:

```
          ID: nf_conntrack
    Function: kmod.present
      Result: False
     Comment: Kernel module nf_conntrack is unavailable
     Started: 00:45:19.472246
    Duration: 29.44 ms
     Changes:
```

### Setup

```yaml
nf_conntrack:
  kmod.present: []
```

```
$ zcat /proc/config.gz | grep ^CONFIG_NF_CONNTRACK=
CONFIG_NF_CONNTRACK=y
```

### Versions Report

```
Salt Version:
           Salt: 2016.11.4

Dependency Versions:
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: Not Installed
      pycparser: 2.10
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.9 (default, Jun 29 2016, 13:08:31)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: debian 8.9
        machine: x86_64
        release: 4.9.41-cloudflare-2017.8.2
         system: Linux
        version: debian 8.9
```

### PR DESCRIPTION
**Pull request by bobrik**

### CODE REVIEW

**Comment by bobrik** [comment]
https://www.kernel.org/doc/Documentation/kbuild/kbuild.txt

```
modules.builtin
--------------------------------------------------
This file lists all modules that are built into the kernel. This is used
by modprobe to not fail when trying to load something builtin.
```

File: salt/modules/kmod.py [hunk 1]
```diff
@@ -123,7 +123,16 @@ def available():
         salt '*' kmod.available
     '''
     ret = []
+
     mod_dir = os.path.join('/lib/modules/', os.uname()[2])
+
+    built_in_file = os.path.join(mod_dir, 'modules.builtin')
+    if os.path.exists(built_in_file):
+        with open(built_in_file, 'r') as f:
```
Comment (cachedout) [inline salt/modules/kmod.py]: Please replace this with `salt.utils.fopen`

File: salt/modules/kmod.py [hunk 2]
```diff
@@ -123,7 +123,16 @@ def available():
...
+            for line in f:
+                # Strip .ko from the basename
+                ret.append(os.path.basename(line)[:-4])
```
Comment (cachedout) [inline salt/modules/kmod.py]: is `-4` correct here? I assume you just want to strip off the `.ko` on the end, yes?

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/modules/kmod.py (hunk 2 above)
Comment (bobrik) [reply salt/modules/kmod.py]: -4 is correct: `.ko\n`. I can run `.rstrip()` on basename if that makes it easier to understand.

Comment (cachedout) [reply salt/modules/kmod.py]: Missed the newline! My fault. That makes sense.

File: salt/modules/kmod.py (hunk 1 above)
Comment (bobrik) [reply salt/modules/kmod.py]: Done.

**Review by cachedout — APPROVED** [review]

**Comment by rallytime** [comment]
Fixes #43945

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
Test failure is not related.

Thanks @bobrik!

**Pull request merged**